substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/NativeImageOptions.java

```diff
@@ -176,7 +176,7 @@ public boolean compatibleWith(CStandards standard) {
 
     public static CStandards getCStandard() {
         try {
-            return CStandards.valueOf(CStandard.getValue());
+            return CStandards.valueOf(CStandard.getValue().toUpperCase().trim().replace("\"", "").replace("'", ""));
```

Comment:
one line change; for the sake of #6699, non-character/number inputs could simply be rejected, with a message not to include quotes. this felt simpler and more obvious, and with less risk, as the `CStandards` enum is unlikely to change often, but I can easily push a different PR if needed.